packages/babel/src/index.js

```diff
@@ -122,21 +122,29 @@ function createBabelInputPluginFactory(customCallback = returnObject) {
         let exclude;
         let include;
         let extensions;
+        let customFilter;
 
         ({
           exclude,
           extensions,
           babelHelpers,
           include,
+          filter: customFilter,
           skipPreflightCheck,
           ...babelOptions
         } = unpackInputPluginOptions(pluginOptionsWithOverrides, this.meta.rollupVersion));
 
         const extensionRegExp = new RegExp(
           `(${extensions.map(escapeRegExpCharacters).join('|')})$`
         );
-        const includeExcludeFilter = createFilter(include, exclude);
-        filter = (id) => extensionRegExp.test(stripQuery(id).bareId) && includeExcludeFilter(id);
+        const userDefinedFilter = (id) => {
```

Comment:
Sorry for not mentioning that before (I would sweat that I've done that but apparently I didn't) - I would not try to compose those things here. If a custom `filter` gets passed in I would just throw if `exclude` or `include` would be passed in as well. 

Something along those lines:
```js
if (customFilter && (include || exclude)) {
  throw new Error('...')
}
const configFilter = customFiler || createFilter(include, exclude);
filter = (id) => extensionRegExp.test(stripQuery(id).bareId) && configFilter(id);
```

Reply:
Change the code logic, switch to your design. and add test as well.